job_executor/worker/steps/dataset_pseudonymizer.py

```diff
@@ -20,192 +26,136 @@ def _get_unit_types(
+    value_pseudonyms = _pseudonymize_column(
+        input_dataset, "value", measure_unit_id_type, job_id
     )
-    target_file = open(output_csv_path, "w", newline="", encoding="utf-8")
-    with open(input_csv_path, newline="", encoding="utf-8") as csv_file:
-        for line in csv_file:
-            row = line.strip().split(";")
-            line_number: int = row[0]
-            unit_id: str = row[1]
-            value: str = row[2]
-            start_date: str = row[3]
-            stop_date: str = row[4]
-            target_file.write(
-                ";".join(
-                    [
-                        str(line_number),
-                        str(identifier_to_pseudonym[unit_id]),
-                        str(value_to_pseudonym[value]),
-                        start_date,
-                        stop_date,
-                    ]
-                )
-                + "\n"
-            )
-    target_file.close()
-    return output_csv_path
 
+    original_column_names = input_dataset.schema.names
 
-def _pseudonymize_csv(
-    input_csv_path: Path,
-    identifier_unit_id_type: Union[str, None],
-    measure_unit_id_type: Union[str, None],
-    job_id: str,
-) -> Path:
-    if identifier_unit_id_type and not measure_unit_id_type:
-        logger.info("Pseudonymizing identifier")
-        return _pseudonymize_identifier_only(
-            input_csv_path, identifier_unit_id_type, job_id
-        )
-    elif measure_unit_id_type and not identifier_unit_id_type:
-        logger.info("Pseudonymizing measure")
-        return _pseudonymize_measure_only(
-            input_csv_path, measure_unit_id_type, job_id
-        )
-    elif identifier_unit_id_type and measure_unit_id_type:
-        logger.info("Pseudonymizing identifier and measure")
-        return _pseudonymize_identifier_and_measure(
-            input_csv_path,
-            identifier_unit_id_type,
-            measure_unit_id_type,
-            job_id,
-        )
-    else:
-        logger.info("No pseudonymization")
-        return input_csv_path
+    # Initialize arrays_to_include with the same length as original columns with None values
+    arrays_to_include = [None] * len(original_column_names)
+
+    # Replace data for pseudonymized columns
+    if unit_id_pseudonyms:
+        unit_id_index = original_column_names.index("unit_id")
+        arrays_to_include[unit_id_index] = unit_id_pseudonyms
 
+    if value_pseudonyms:
+        value_index = original_column_names.index("value")
+        arrays_to_include[value_index] = value_pseudonyms
 
-def run(input_csv_path: Path, metadata: Metadata, job_id: str) -> Path:
+    # Replace None values with original column data
+    for i, column_data in enumerate(arrays_to_include):
+        if column_data is None:
+            arrays_to_include[i] = input_dataset.to_table(
+                columns=[original_column_names[i]]
+            )[original_column_names[i]].chunk(0)
```

Comment:
Could it be that you used chunk(0) here because pyarrow.Table.from_arrays would accept a chunked array? 🤔 If thats the case we need to do this differently, because when you grab the first chunk "0" you might not get the entirety of the column in a bigger dataset

Reply:
Yes you correct, not sure if pyarrow can do this automagic but its an options you can use when reading large datasets.

But it would be safer to avoid it here!